### PR TITLE
Add escrow contract

### LINKED ISSUE #95: Escrow contract
Related: https://github.com/TelediskoDAO/pm/issues/9

### PR DESCRIPTION
**Pull request by vrde**
This is a very simple implementation that can be extended later with more features.
Check https://github.com/TelediskoDAO/contracts/blob/e1f25de92da7aed552cd8b2214e74a8b34a04ee7/contracts/Escrow/Escrow.sol for an explanation on how this works.

Close #95

### CODE REVIEW

File: contracts/Escrow/Escrow.sol [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    constructor(IERC20MatchOffer nativeToken, IERC20 secondaryToken) {
+        _nativeToken = nativeToken;
+        _secondaryToken = secondaryToken;
+    }
+
+    function matchOffer(address from, uint256 amount) public {
+        _secondaryToken.transferFrom(_msgSender(), from, amount);
+        _nativeToken.matchOffer(from, _msgSender(), amount);
+    }
```
Comment (sirnicolaz) [inline contracts/Escrow/Escrow.sol:52]: To make it more clear, would it be possible to rename things this way:
```
 function swap(address nativeTokenOwner, uint256 amount) public {
        _secondaryToken.transferFrom(_msgSender(), nativeTokenOwner, amount);
        _nativeToken.matchOffer(nativeTokenOwner, _msgSender(), amount);
}
```

The name of the method reflects better what the `Escrow` itself is actually doing and also avoid having the possibly confusing `transferFrom(_msgSender(), from, amount)`

File: test/Escrow.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    // Allow escrow contract to withdraw funds
+    await stableTokenMock.connect(contributor).approve(escrow.address, 100);
+
+    // Match the offer
+    await expect(
+      escrow.connect(contributor).matchOffer(contributor2.address, 20)
+    )
+      .to.emit(stableTokenMock, "Transfer")
+      .withArgs(contributor.address, contributor2.address, 20);
+  });
```
Comment (sirnicolaz) [inline test/Escrow.ts:82]: Would it make sense add an "unhappy path" test?

**Review by sirnicolaz — CHANGES_REQUESTED** [review]
A couple of minor issues, but overall good and simple solution. One missing piece is the deployment script for this type of contract.

**Comment by vrde** [comment]
Replaced by #115

**Pull request closed without merging**